### PR TITLE
Implement DamageType.Builder

### PR DESCRIPTION
**Pull request by ItsDoot**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/2271) | **Sponge**

### CODE REVIEW

File: src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.spongepowered.common.event.cause.entity.damage;
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java:25]: Tsk

File: src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.spongepowered.api.ResourceKey;
+import org.spongepowered.api.event.cause.entity.damage.DamageType;
+import org.spongepowered.common.util.SpongeCatalogBuilder;
+
+public class SpongeDamageTypeBuilder extends SpongeCatalogBuilder<DamageType, DamageType.Builder> implements DamageType.Builder {
+
+    @Override
+    protected DamageType build(ResourceKey key) {
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java]: Tsk tsk

File: src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java (hunk 1 above)
Comment (ItsDoot) [reply src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java:25]: Fixed

File: src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java (hunk 2 above)
Comment (ItsDoot) [reply src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java]: Made final 👀

File: src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.event.cause.entity.damage;
+
+import org.spongepowered.api.ResourceKey;
+import org.spongepowered.api.event.cause.entity.damage.DamageType;
+import org.spongepowered.common.util.SpongeCatalogBuilder;
+
+public class SpongeDamageTypeBuilder extends SpongeCatalogBuilder<DamageType, DamageType.Builder> implements DamageType.Builder {
```
Comment (dualspiral) [inline src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java]: Before @Zidane gets to it...

Comment (ItsDoot) [reply src/main/java/org/spongepowered/common/event/cause/entity/damage/SpongeDamageTypeBuilder.java]: Fixed

**Pull request merged**